File: atlas/src/atlas-cli/src/application/federation/model/AtlasWorkspaceManifest.ts
/**
 * Selects the independently generated module models that form one federated Atlas workspace.
 */
export interface AtlasWorkspaceManifest {
  /** The independently versioned workspace-manifest schema. */
  readonly schemaVersion: 1;
  /** Manifest entries in deterministic artifact-ID order. */
  readonly modules: readonly AtlasWorkspaceManifestEntry[];
}

/**
 * Locates one model file and asserts the artifact identity expected by the manifest.
 */
export interface AtlasWorkspaceManifestEntry {
  /** Expected opaque artifact ID. */
  readonly moduleId: string;
  /** Manifest-relative, slash-normalized model path. */
  readonly modelPath: string;
}

File: atlas/src/tools/cs/atlas-cs-sdk/Model/AtlasWorkspaceManifest.cs
namespace StarCruiseStudios.Atlas.Cs.Model;

/// <summary>
/// Lists the portable module models selected for one generated C# workspace.
/// </summary>
/// <param name="SchemaVersion">Workspace manifest schema version; currently 1.</param>
/// <param name="Modules">Module entries sorted by module ID.</param>
public sealed record AtlasWorkspaceManifest(
    int SchemaVersion,
    IReadOnlyList<AtlasWorkspaceManifestEntry> Modules);
